mousebender/simple.py

```diff
@@ -183,14 +200,54 @@ def handle_starttag(self, tag, attrs_list):
         # Links in the simple repository MAY have a data-yanked attribute which
         # may have no value, or may have an arbitrary string as a value.
         yanked = "data-yanked" in attrs, attrs.get("data-yanked") or ""
-
-        self.archive_links.append(
+        self.update_index(
             ArchiveLink(filename, url, requires_python, hash_, gpg_sig, yanked)
         )
 
+    def update_index(self, archive: ArchiveLink):
+        # old way
+        self.archive_links.append(archive)
```

Comment:
I've left this here for now intentionally, while the PR is a _draft_. Will remove and fully update tests once I get some constructive criticism/feedback on my review.